service/history/workflow/mutable_state_impl.go

```diff
@@ -4264,6 +4264,29 @@ func (ms *MutableStateImpl) AddWorkflowExecutionUpdateAdmittedEvent(request *upd
 	return event, nil
 }
 
+func (ms *MutableStateImpl) DeleteSubStateMachine(path *persistencespb.StateMachinePath) error {
+	incomingPath := []hsm.Key{}
```

Comment:
```suggestion
	incomingPath := make([]hsm.Key, len(path))
```

And then you can just assign with the index.

Reply:
updated